tareas/sam_tareas/Tarea-2.jmd

```diff
@@ -0,0 +1,542 @@
+
+(b) Repitan el ejercicio anterior para el mapeo $G(x) = 2x\mod 1$, definido en el intervalo $x\in[0,1]$.
+
+(c) ¿Qué conclusión podemos sacar de los histogramas para ambos mapeos?
+
+```julia
+# Histograma para x^2-2 
+its, its_x, its_y =itera_mapeo(x-> x^2 -2,0.33,50000)
+histogram(its, nbins=100,color= "pink", title="f(x) = x^2 - 2",legend=false,normed= true)
+```
+
+```julia
+#Histograma de x^2 - 2 para  un punto fijo 
+its, its_x, its_y =itera_mapeo(x-> x^2-2 ,2,50000)
+histogram(its, nbins=50,color= "pink", title="f(x) = x^2-2",normed= true,legend=false)
+```
+
+```julia
+# Histograma para  2x mod(1)
+its, its_x, its_y =itera_mapeo(x->2x %1  ,0.0,100000)
+histogram(its, nbins=50,color= "red", title="f(x) = 2x mod 1",normed= true,legend=false)
+```
+
+```julia
+# Histograma para  2x mod(1)
+its, its_x, its_y =itera_mapeo(x->2x %1  ,0.165,100000)
+histogram(its, nbins=50,color= "red", title="f(x) = 2x mod 1",normed= true,legend=false)
+```
+
+En el inciso (a) se puede observar que la mayoría de los puntos están en -2 y 2 para un punto inicial aleatorio en ese intervalo , por lo que se ve que la órbita de la función oscila más en esos puntos . Pero al plotear el punto fijo se tiene toda la pantalla del 2 al 3 , algo que es raro ya que en el punto fijo la órbita es de periódo 1. 
+En el inciso (b) ocurre lo mismo para el punto fijo , pero para cualquier punto entre 0 y 1 se tiene que la órbita solo oscila cerca del cero.
+
+## Ejercicio 3
+
+(a) Usando lo que hicieron en la Tarea 1, incluyan lo que desarrollaron para los números `Dual`es en un módulo `NumDual` de Julia ([ver la documentación aquí](https://docs.julialang.org/en/v1.3/manual/modules/)). En particular, el módulo debe exportar el tipo `Dual` y la función `var_dual`, al menos. El archivo con el módulo lo deben incluir en un archivo ".jl" en su propio directorio de tareas. Carguen el módulo en este notebook, usando
+
+```julia
+include("NumDual.jl")
+```
+
+```julia
+using Main.NumDual
+```
+
+(b) Escriban una función que implemente el método de Newton para funciones en una dimensión. La derivada que se requiere debe ser calculada a través de los números duales. Obtengan un cero de $f(x) = x^3 - 8$, como test de que la implementación es correcta.
+
+(c) Encuentren los puntos fijos del mapeo $F(x) = x^2 - 1.1$ usando la función que implementaron para el método de Newton.
+
+(d) Encuentren las puntos *de periodo 2* para el mapeo $F(x) = x^2 - 1.1$ usando la función que implementaron para el método de Newton.
+
+(e) Usen números duales para mostrar que los puntos de periodo 2 para el mapeo $F(x) = x^2 -1$ son linealmente estables (atractivos).
+
+```julia
+function Newton(f::Function,x0,n)     
+  x1 = x0 + 1  
+    for i= 1:n
+        x1 = x0 - f(var_dual(x0)).x/f(var_dual(x0)).y
```

Comment:
Esta línea conviene escribirla de la siguiente manera:
```julia
f_dual = f(var_dual(x0))
x1 = x0 - f_dual.x/f_dual.y
```
La ventaja de hacerlo así es que sólo haces una evaluación de `f(var_dual(x0))` en lugar de dos que es como lo tienes, y usas ese resultado dos veces.

Esta celda me da un error, dado que en el módulo de `NumDual` no exportaste `var_dual`.

Reply:
![Screenshot (207)](https://user-images.githubusercontent.com/61710178/79192920-78231c00-7dde-11ea-9812-809f122aa8f0.png)
Lo malo de hacerlo así es que surge un error para calcular las raíces y no la calcula para números cercanos, y da números muy lejanos y de la otra forma no...

Reply:
Lo  que ocurre es que  sacaste `f_dual` del ciclo `for`. La diferencia es que sólo usas un valor de la función y su derivada, en lugar de "iterar" el punto y recalcular la función y su derivada en los iterados.